newdocs/docs/api/rasa-sdk.mdx

```diff
@@ -0,0 +1,359 @@
+that both need to be overwritten, `name()` and `run()`.
+
+<a aria-hidden="true" tabIndex="-1" className="anchor enhancedAnchor" id="custom-action-example"></a>
+
+In a restaurant bot, if the user says “show me a Mexican restaurant”,
+your bot could execute the action `ActionCheckRestaurants`,
+which might look like this:
+
+```python3
+from rasa_sdk import Action
+from rasa_sdk.events import SlotSet
+
+class ActionCheckRestaurants(Action):
+   def name(self) -> Text:
+      return "action_check_restaurants"
+
+   def run(self,
+           dispatcher: CollectingDispatcher,
+           tracker: Tracker,
+           domain: Dict[Text, Any]) -> List[Dict[Text, Any]]:
+
+      cuisine = tracker.get_slot('cuisine')
+      q = "select * from restaurants where cuisine='{0}' limit 1".format(cuisine)
+      result = db.query(q)
+
+      return [SlotSet("matches", result if result is not None else [])]
+```
+
+You should add the action name `action_check_restaurants` to
+the actions in your domain file. The action’s `run()` method receives
+three arguments. You can access the values of slots and the latest message
+sent by the user using the `tracker` object, and you can send messages
+back to the user with the `dispatcher` object, by calling
+`dispatcher.utter_message`.
+
+Details of the `run()` method:
+
+
+#### Action.run
+async Action.rundispatcher, tracker, domain**:
+Execute the side effects of this action.
+
+
+* **Parameters**
+
+  * **dispatcher** – the dispatcher which is used to
+    send messages back to the user. Use
+    `dipatcher.utter_message()` or any other
+    `rasa_sdk.executor.CollectingDispatcher`
+    method.
+
+  * **tracker** – the state tracker for the current
+    user. You can access slot values using
+    `tracker.get_slot(slot_name)`, the most recent user message
+    is `tracker.latest_message.text` and any other
+    `rasa_sdk.Tracker` property.
+
+  * **domain** – the bot’s domain
+
+
+
+* **Returns**
+
+  A dictionary of `rasa_sdk.events.Event` instances that is
+
+    returned through the endpoint
+
+
+
+
+* **Return type**
+
+  `List`[`Dict`[`str`, `Any`]]
+
+
+Details of the `dispatcher.utter_message()` method:
+
+
+#### CollectingDispatcher.utter_message
+CollectingDispatcher.utter_messagetext=None, image=None, json_message=None, template=None, attachment=None, buttons=None, elements=None, \*\*kwargs**:
+“Send a text to the output channel
+
+
+* **Return type**
+
+  `None`
+
+
+<a aria-hidden="true" tabIndex="-1" className="anchor enhancedAnchor" id="custom-session-start"></a>
+
+### Customizing the session start action
+
+The default behavior of the session start action is to take all existing slots and to
+carry them over into the next session. Let’s say you do not want to carry over all
+slots, but only a user’s name and their phone number. To do that, you’d override the
+`action_session_start` with a custom action that might look like this:
+
+```python3
```

Comment:
we should use `python` instead (see [this](https://github.com/FormidableLabs/prism-react-renderer/blob/11b33868c9b83d31687ea09baf944de07366de1f/src/vendor/prism/includeLangs.js#L29))

Reply:
let's do this manually